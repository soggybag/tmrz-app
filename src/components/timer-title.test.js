import TimerTitle from './timer-title'
import Timer from '../reducers/timer'

// https://www.leighhalliday.com/testing-react-jest-enzyme-sinon

describe('TimerTitle', () => {
  it ('should render correctly', () => {
    const name = "Hello";
    const timer = new Timer(name)
    // timer.time = 12345
    const wrapper = shallow(<TimerTitle timer={timer} />)
    expect(wrapper).toMatchSnapshot()
  })
})

// import React from 'react';
// import { shallow, configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// configure({ adapter: new Adapter() });
//
// import TimerTitle from './timer-title';
// import Timer from '../reducers/timer';
//
describe('TimerTitle 2', () => {
  it('Should render', () => {
    const name = "Hello";
    const timer = new Timer(name);
    const timerTitle = shallow(<TimerTitle timer={timer} />);
    expect(timerTitle.find('.timer-view-name__h2').length).toBe(1);
    expect(timerTitle.find('.timer-view-name__h2').text()).toEqual(name);
    expect(timerTitle.find('.timer-view-time__h1').length).toBe(1);
    expect(timerTitle.find('.timer-view-time__h1').text()).toEqual('0:0:0.0');
  });
});





//
