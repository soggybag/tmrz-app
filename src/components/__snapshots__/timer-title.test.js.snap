// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerTitle should render correctly 1`] = `
<div>
  <h2
    className="timer-view-name__h2"
  >
    Hello
  </h2>
  <h1
    className="timer-view-time__h1"
  >
    0:0:0.0
  </h1>
  <small>
    0
  </small>
</div>
`;
